### PR TITLE
rove lock in crd controller

### PR DESCRIPTION
**Pull request by ldpliu**
## Related issue(s)

Fixes #

## Tests
* [ ] Unit/function tests have been added and incorporated into `make unit-tests`.
* [ ] Integration tests have been added and incorporated into `make integration-test`.
* [ ] E2E tests have been added and incorporated into `make e2e-test-all`.
* [ ] List other manual tests you have done.

### CODE REVIEW

File: operator/pkg/controllers/crd/crd_controller.go [hunk 1]
```diff
@@ -187,11 +183,10 @@ func AddCRDController(mgr ctrl.Manager, operatorConfig *config.OperatorConfig,
 
 	crdPred := predicate.Funcs{
 		CreateFunc: func(e event.CreateEvent) bool {
-			// the predicate function might be invoked in different controller goroutines.
-			controller.mu.Lock()
-			_, ok := controller.resources[e.Object.GetName()]
-			controller.mu.Unlock()
-			return ok
+			if KafkaCrds.Has(e.Object.GetName()) || ACMCrds.Has(e.Object.GetName()) {
```
Comment (clyang82) [inline operator/pkg/controllers/crd/crd_controller.go:186]: just `return KafkaCrds.Has(e.Object.GetName()) || ACMCrds.Has(e.Object.GetName())`

Comment (ldpliu) [reply operator/pkg/controllers/crd/crd_controller.go]: > simplify to `return KafkaCrds.Has(e.Object.GetName()) || ACMCrds.Has(e.Object.GetName())`?

done

**Pull request merged**